frontend/tests/tests.test.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable */
+/* todo's: optimizations
+    - Running on different deployments
+    (a) diff file for each. maybe a commmon test file which is a function that takes appUrl, and all that local.test.js would do was set appUrl and then pass it to that func. Then could run via `npx playwright test FILENAME` and make a `make` command as shorthand
+    (b) Loop: 1 file and loop over each deployment or pass a flag to select which one - (what's implemented here)
+    (c) Command?: make a command that sets the URL in a file and then import that into the playwright test
+    (d) playwright.config.js? - (didn't work; kinda makes sense since that url doesn't get passed)
+ */
+/*
+ */
+// @ts-check
+const { test, expect } = require('@playwright/test');
+
+const deploymentConfigs = {
+  local: 'http://127.0.0.1:3000',
+  // dev: 'https://icy-ground-0416a040f.2.azurestaticapps.net
+  // prod: 'https://purple-plant-0f4023d0f.2.azurestaticapps.net'
+};
+
+for (const envName in deploymentConfigs) {
+  const appUrl = deploymentConfigs[envName];
+  test(envName + ': ' + 'Main page - has title & heading', async ({ page }) => {
+    await page.goto(appUrl);
+    await expect(page).toHaveTitle(/TermHub/);  // Expect a title "to contain" a substring.
+    // todo: the version is going to change. need to account for that.
+    await expect(page.getByRole('heading', { name: 'Welcome to TermHub! Beta version 0.3.1' })).toBeVisible();
+  });
+
+  test(envName + ': ' + 'Help / about - hyperlink & title', async ({ page }) => {
+    await page.goto(appUrl);
+    await page.getByRole('link', { name: 'Help / About' }).click();
+    await expect(page.getByRole('heading', { name: 'About TermHub\n' })).toBeVisible();
+  });
+
+  // todo: rename this as I add onto it. I think I want to do the workflow from search -> comparison
+  test(envName + ': ' + 'Cset search - select, load, compare', async ({ page }) => {
+    const testCodesetId = '1000002363';
+    // Load page and click menu
+    await page.goto(appUrl);
+
+    // close alert panel if it appears
+    const alertPanelClose = await page.waitForSelector('[data-testid=flexcontainer-Alerts] button');
+    if (alertPanelClose) {
+      await alertPanelClose.click();
+    }
+
+    // todo: alternative: data-testid="autocomplete"
+    const searchWidget = await page.waitForSelector('#add-codeset-id');
```

Comment:
Hmm, I don't know why this is failing now, timing out and unable to find this widget.

```
    Test timeout of 30000ms exceeded.

    Error: page.waitForSelector: Page closed
    =========================== logs ===========================
    waiting for locator('#add-codeset-id') to be visible
    ============================================================

      46 |
      47 |     // todo: alternative: data-testid="autocomplete"
    > 48 |     const searchWidget = await page.waitForSelector('#add-codeset-id');
```

Reply:
oh right, the backend wasn't running